http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -222,14 +228,15 @@ protected String getRedirectUri(String state) {
             String redirectUri = rewrittenRedirectUri(url);
             URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl());
             redirectUriBuilder.addParameter(RESPONSE_TYPE, CODE)
-                    .addParameter(CLIENT_ID, deployment.getResourceName());
+                    .addParameter(CLIENT_ID, deployment.getResourceName())
+                    .addParameter(NONCE, sessionRandomValueHash);
```

Comment:
I don't think we should do this here. I just realized that for the REQUEST and REQUEST_URI cases below, this will result in the nonce being added to both the redirectUriBuilder and to the JWT claims. I think `addParameter(NONCE, sessionRandomValueHash);` should be called from createOAuthRequest instead.

Reply:
done